--- daemons/Makefile
+++ daemons/Makefile
@@ -29,7 +29,7 @@ HURDLIBS = fshelp ports shouldbeinlibc
 OBJS = $(SRCS:.c=.o)
 getty-LDLIBS = -lutil
 
-INSTALL-mail.local-ops = -o root -m 4755
+INSTALL-mail.local-ops = -m 4755
 
 include ../Makeconf
 
--- utils/Makefile
+++ utils/Makefile
@@ -42,10 +42,10 @@ setauth-LDLIBS = $(LIBCRYPT)
 mount-LDLIBS = $(libblkid_LIBS)
 mount-CPPFLAGS = $(libblkid_CFLAGS)
 
-INSTALL-login-ops = -o root -m 4755
-INSTALL-ids-ops = -o root -m 4755
-INSTALL-ps-ops = -o root -m 4755
-INSTALL-w-ops = -o root -m 4755
+INSTALL-login-ops = -m 4755
+INSTALL-ids-ops = -m 4755
+INSTALL-ps-ops = -m 4755
+INSTALL-w-ops = -m 4755
 
 include ../Makeconf
 
