Don't override the environment settings.

--- Makefile
+++ Makefile
@@ -1,24 +1,15 @@
 # Unix makefile for JBIG-KIT
 
-# Select an ANSI/ISO C compiler here, GNU gcc is recommended
-CC = gcc
-
-# Options for the compiler: A high optimization level is suggested
-CFLAGS = -O2 -W -Wno-unused-result
-# CFLAGS = -O -g -W -Wall -Wno-unused-result -ansi -pedantic # -DDEBUG
-
-export CC CFLAGS
-
 VERSION=2.1
 
 all: lib pbm
 	@echo "Enter 'make test' in order to start some automatic tests."
 
 lib:
-	cd libjbig && $(MAKE) -e
+	make -C libjbig "VERSION=$(VERSION)"
 
 pbm: lib
-	cd pbmtools && $(MAKE) -e
+	make -C pbmtools
 
 test: lib pbm
 	cd libjbig  && $(MAKE) test
--- libjbig/Makefile
+++ libjbig/Makefile
@@ -1,11 +1,5 @@
 # Unix makefile for the JBIG-KIT library
 
-# Select an ANSI/ISO C compiler here, GNU gcc is recommended
-CC = gcc
-
-# Options for the compiler: A high optimization level is suggested
-CFLAGS = -g -O -W -Wall -ansi -pedantic # --coverage
-
 all: libjbig.a libjbig85.a tstcodec tstcodec85
 
 tstcodec: tstcodec.o jbig.o jbig_ar.o
@@ -16,13 +10,13 @@
 
 libjbig.a: jbig.o jbig_ar.o
 	rm -f libjbig.a
-	ar rc libjbig.a jbig.o jbig_ar.o
-	-ranlib libjbig.a
+	$(AR) rc libjbig.a jbig.o jbig_ar.o
+	$(RANLIB) libjbig.a
 
 libjbig85.a: jbig85.o jbig_ar.o
 	rm -f libjbig85.a
-	ar rc libjbig85.a jbig85.o jbig_ar.o
-	-ranlib libjbig85.a
+	$(AR) rc libjbig85.a jbig85.o jbig_ar.o
+	$(RANLIB) libjbig85.a
 
 jbig.o: jbig.c jbig.h jbig_ar.h
 jbig85.o: jbig85.c jbig85.h jbig_ar.h
--- pbmtools/Makefile
+++ pbmtools/Makefile
@@ -1,10 +1,5 @@
 # Unix makefile for the JBIG-KIT PBM tools
 
-# Select an ANSI/ISO C compiler here, e.g. GNU gcc is recommended
-CC = gcc
-
-# Options for the compiler
-CFLAGS = -g -O -W -Wall -Wno-unused-result -ansi -pedantic # --coverage
 CPPFLAGS = -I../libjbig 
 
 .SUFFIXES: .1 .5 .txt $(SUFFIXES)
