This links a missing library onto mpicalc.

--- src/Makefile.am
+++ src/Makefile.am
@@ -132,7 +132,7 @@
 
 mpicalc_SOURCES = mpicalc.c
 mpicalc_CFLAGS = $(GPG_ERROR_CFLAGS)
-mpicalc_LDADD = libgcrypt.la $(GPG_ERROR_LIBS) $(LIBTHREAD)
+mpicalc_LDADD = libgcrypt.la $(GPG_ERROR_LIBS) $(LIBTHREAD) -ldl
 
 hmac256_SOURCES = hmac256.c
 hmac256_CFLAGS = -DSTANDALONE $(arch_gpg_error_cflags)
--- src/Makefile.in
+++ src/Makefile.in
@@ -512,7 +512,7 @@
 dumpsexp_LDADD = $(arch_gpg_error_libs) $(LIBTHREAD)
 mpicalc_SOURCES = mpicalc.c
 mpicalc_CFLAGS = $(GPG_ERROR_CFLAGS)
-mpicalc_LDADD = libgcrypt.la $(GPG_ERROR_LIBS) $(LIBTHREAD)
+mpicalc_LDADD = libgcrypt.la $(GPG_ERROR_LIBS) $(LIBTHREAD) -ldl
 hmac256_SOURCES = hmac256.c
 hmac256_CFLAGS = -DSTANDALONE $(arch_gpg_error_cflags)
 hmac256_LDADD = $(arch_gpg_error_libs) $(LIBTHREAD)
