Put the seed file in a more appropriate location.

--- ChangeLog
+++ ChangeLog
@@ -3999,7 +3999,7 @@
 	* src/testsuite/conv-3-test: New test program.
 
 	* contrib/lshd.debian.init: Check for existance of
-	/var/spool/lsh/yarrow-seed-file.
+	/var/lib/lsh/yarrow-seed-file.
 	(restart): New target, which sends SIGHUP to the running lshd and
 	then starts a new one.
 
--- contrib/lshd.debian.init
+++ contrib/lshd.debian.init
@@ -26,7 +26,7 @@
 if [ ! -x $DAEMON \
      -o ! -f /etc/lsh_host_key  \
      -o ! -f /etc/lsh_host_key.pub \
-     -o ! -f /var/spool/lsh/yarrow-seed-file ]
+     -o ! -f /var/lib/lsh/yarrow-seed-file ]
 then
 	echo "lshd not installed properly" >&2
         exit 1
--- contrib/solpkg.sh.in
+++ contrib/solpkg.sh.in
@@ -67,7 +67,7 @@
 echo Creating random seed
 #!/bin/sh
 @bindir@/lsh-make-seed --server 
-if [ -f /var/spool/lsh/yarrow-seed-file -a \! -f /etc/lsh_host_key ] ; then
+if [ -f /var/lib/lsh/yarrow-seed-file -a \! -f /etc/lsh_host_key ] ; then
   echo Creating host key
   @bindir@/lsh-keygen --server | @bindir@/lsh-writekey --server 
 else
--- doc/lsh.html
+++ doc/lsh.html
@@ -608,7 +608,7 @@
 <pre class="example">     lsh-make-seed --server
 </pre>
    <p>as root. The seed file is stored as
-<samp><span class="file">/var/spool/lsh/yarrow-seed-file</span></samp>.
+<samp><span class="file">/var/lib/lsh/yarrow-seed-file</span></samp>.
 
 <div class="node">
 <a name="lsh-basics"></a>
@@ -1679,7 +1679,7 @@
 
      <br><dt><samp><span class="file">/etc/lsh_host_key.pub</span></samp><dd>The corresponding public key.
 
-     <br><dt><samp><span class="file">/var/spool/lsh/yarrow-seed-file</span></samp><dd>The seed-file for <samp><span class="command">lshd</span></samp>'s randomness generator.
+     <br><dt><samp><span class="file">/var/lib/lsh/yarrow-seed-file</span></samp><dd>The seed-file for <samp><span class="command">lshd</span></samp>'s randomness generator.
 
      <br><dt><samp><span class="file">~/.lsh/authorized_keys</span></samp><dd>This is a directory that keeps a &ldquo;database&rdquo; of keys authorized for
 login. With the current implementation, a key is authorized for login
--- doc/lsh.info
+++ doc/lsh.info
@@ -478,7 +478,7 @@
      lsh-make-seed --server
 
    as root. The seed file is stored as
-`/var/spool/lsh/yarrow-seed-file'.
+`/var/lib/lsh/yarrow-seed-file'.
 
 
 File: lsh.info,  Node: lsh basics,  Next: tcpip forwarding,  Prev: lsh-make-seed,  Up: Getting started
@@ -1503,7 +1503,7 @@
 `/etc/lsh_host_key.pub'
      The corresponding public key.
 
-`/var/spool/lsh/yarrow-seed-file'
+`/var/lib/lsh/yarrow-seed-file'
      The seed-file for `lshd''s randomness generator.
 
 `~/.lsh/authorized_keys'
--- doc/lsh.texinfo
+++ doc/lsh.texinfo
@@ -554,7 +554,7 @@
 @end example
 
 as root. The seed file is stored as
-@file{/var/spool/lsh/yarrow-seed-file}.
+@file{/var/lib/lsh/yarrow-seed-file}.
 
 
 @node lsh basics, tcpip forwarding, lsh-make-seed , Getting started
@@ -1679,7 +1679,7 @@
 @item /etc/lsh_host_key.pub
 The corresponding public key.
 
-@item /var/spool/lsh/yarrow-seed-file
+@item /var/lib/lsh/yarrow-seed-file
 The seed-file for @command{lshd}'s randomness generator.
 
 @item ~/.lsh/authorized_keys
--- doc/lshd.8
+++ doc/lshd.8
@@ -207,7 +207,7 @@
 seed file and the server key.
 By default 
 
-/var/spool/lsh/yarrow-seed-file 
+/var/lib/lsh/yarrow-seed-file 
 
 is used as random seed file (see 
 .B ENVIRONMENT
--- src/environ.h.in
+++ src/environ.h.in
@@ -17,7 +17,7 @@
 #define PATH_EXECUV		SBINDIR "/lsh-execuv"
 
 /* Some other files: /dev/null, /tmp/.X11-unix..., /var/run, /, /tmp,
- * /var/spool/lsh/yarrow-seed-file, /etc/lsh_host_key,
+ * /var/lib/lsh/yarrow-seed-file, /etc/lsh_host_key,
  * /var/run/lsh_proxy.pid, /var/run/lshd.pid, /dev/ptmx, /dev/ptyxx,
  * /dev/ttyxx, /tmp/.X11-unix, /var/adm/wtmp, /var/adm/wtmpx,
  * /bin/sh. */
--- src/lsh-make-seed.c
+++ src/lsh-make-seed.c
@@ -210,8 +210,8 @@
 	argp_error(state, "You can use at most one -o or --server option.");
       else
 	{
-	  self->directory = make_string("/var/spool/lsh");
-	  self->filename =  make_string("/var/spool/lsh/yarrow-seed-file");
+	  self->directory = make_string("/var/lib/lsh");
+	  self->filename =  make_string("/var/lib/lsh/yarrow-seed-file");
 	}
       break;
       
--- src/unix_random.c
+++ src/unix_random.c
@@ -471,7 +471,7 @@
 
   /* FIXME: What's a proper place for this? */
   file_name = make_string(env_name ? env_name
-			  : "/var/spool/lsh/yarrow-seed-file");
+			  : "/var/lib/lsh/yarrow-seed-file");
 
   r = random_init(file_name);
   
